hawkular-tenant-jaxrs-filter/src/main/java/org/hawkular/jaxrs/filter/tenant/TenantFeature.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.jaxrs.filter.tenant;
+
+import java.lang.reflect.Method;
+
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * @author Juraci Paixão Kröhling
+ */
+@Provider
+public class TenantFeature implements DynamicFeature {
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
+        Class<?> resourceClass = resourceInfo.getResourceClass();
+        Method method = resourceInfo.getResourceMethod();
+
+        boolean required = true;
+        if (resourceClass.isAnnotationPresent(TenantRequired.class)) {
+            required = resourceClass.getAnnotation(TenantRequired.class).value();
+        }
+
+        if (method.isAnnotationPresent(TenantRequired.class)) {
```

Comment:
While it might not be the best Maven practice (I honestly don't know and believe your expertise here), it's been used in Hawkular all over the place:
`find -name 'pom.xml' | xargs grep '${project.version}' | wc -l` in various components:

metrics: 42
inventory: 60
alerts: 33